### PR TITLE
Feature/setup filter test data

### PR DESCRIPTION
**Pull request by nshumoogum**
### What

Rewrite of filter tests to teardown and setup test data, instead of removing all data at the beginning of running test suite. Requires https://github.com/ONSdigital/dp-filter-api/pull/30 to be merged

### How to review

Run filter API tests by doing the following:

1) Run dataset API and filter API
2) cd dp-api-tests/filterAPI
3) go test ./...

Check logic in tests

### Who can review

Team A, B or @pathimanoj

### CODE REVIEW

File: filterAPI/delete_dimension_option_test.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestSuccessfullyDeleteRemoveDimensionOptions(t *testing.T) {
+
+	if err := mongo.Teardown(database, collection, "_id", filterID); err != nil {
+		os.Exit(1)
+	}
+
+	if err := setupInstance(); err != nil {
```
Comment (Matt-Guest) [inline filterAPI/delete_dimension_option_test.go]: Not all tests need an instance in mongo

File: filterAPI/delete_dimension_option_test.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+		})
+	})
+
+	Convey("Given a filter job", t, func() {
+
+		if err := mongo.Setup(database, collection, "_id", filterID, ValidFilterJobWithMultipleDimensions); err != nil {
+			os.Exit(1)
+		}
+
+		// TODO Reinstate commented out code below once API has been updated
```
Comment (Matt-Guest) [inline filterAPI/delete_dimension_option_test.go]: Is this a bug which has been raised?

File: filterAPI/delete_dimension_option_test.go [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+			timeDimResponse.Element(3).Object().Value("option").NotEqual("April 1997").Equal("December 1997")
+		})
+	})
+
+	if err := teardownInstance(); err != nil {
+		log.ErrorC("Unable to teardown instance", err, nil)
+		os.Exit(1)
+	}
+
+	if err := mongo.Teardown(database, collection, "_id", filterID); err != nil {
```
Comment (Matt-Guest) [inline filterAPI/delete_dimension_option_test.go]: Can you also log out the err here?

File: filterAPI/initialise.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+	"github.com/ONSdigital/go-ns/log"
+)
+
+var cfg *config.Config
+
+const (
+	database    = "filters"
+	collection  = "filters"
+	instanceID  = "789"
+	filterID    = "123"
```
Comment (Matt-Guest) [inline filterAPI/initialise.go]: Could we generate the instanceID and filterID when the tests run? So if two tests run against the same database they won't override each other.

File: filterAPI/post_dimension_test.go [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+		os.Exit(1)
+	}
+
+	if err := teardownInstance(); err != nil {
+		log.ErrorC("Unable to teardown instance", err, nil)
+		os.Exit(1)
+	}
+}
+
+// func TestFailureToPostDimension(t *testing.T) {
```
Comment (Matt-Guest) [inline filterAPI/post_dimension_test.go]: Is this test needed?

Comment (nshumoogum) [reply filterAPI/post_dimension_test.go]: Yes, it shouldn't be commented out.

**Review by Matt-Guest — APPROVED** [review]
Looks great and all the tests passed on my mac 👍

**Pull request merged**